File: rust-workspace/libs/corelib/src/domain/shared/event.rs
use serde::{Deserialize, Serialize};

use crate::domain::shared::value_object::{EventId, OcurredAt};

#[derive(Debug, Serialize, Deserialize,Clone)]
pub struct EventMeta {
    pub event_id: EventId,
    pub occurred_at: OcurredAt,
    pub aggregate_id: String,
    pub aggregate_version: u64,
}

File: node-workspace/libs/corelib/src/domain/shared/event.ts
import { IdentityId } from "../identity/value-objects/identity_id";
import { OcurredAt } from "../shared/value-objects/ocurred_at";
import { EventId } from "./value-objects/event_id";

export interface EventMeta {
    event_id: EventId;
    occurred_at: OcurredAt;
    aggregate_id: string | IdentityId;
    aggregate_version: number;
}

export function makeEventMeta(args: {
    event_id: EventId;
    occurred_at: OcurredAt;
    aggregate_id: string;
    aggregate_version: number;
}): EventMeta {
    return { ...args };
}
